projects/plugins/jetpack/modules/google-fonts.php

```diff
@@ -115,6 +132,63 @@ function jetpack_add_google_fonts_provider() {
 }
 add_action( 'after_setup_theme', 'jetpack_add_google_fonts_provider' );
 
+/**
+ * Updates font names by filtering the data provided by the theme for global styles & settings.
+ *
+ * `wp_register_webfonts()` does not currently support giving a separate "name"
+ * from "font-family", even if theme.json schema itself does support it.
+ * See: https://github.com/WordPress/gutenberg/issues/46398
+ *
+ * Once support for "name" is added, below functionality can be refactored
+ * to provide the name directly within `wp_register_webfonts()`.
+ *
+ * @param WP_Theme_JSON_Data_Gutenberg $theme_json Class to access and update the underlying data.
+ * @return WP_Theme_JSON_Data_Gutenberg Class with updated Global Styles settings.
+ */
+function jetpack_rename_google_font_names( $theme_json ) {
+	$raw_data      = $theme_json->get_data();
+	$font_families = $raw_data['settings']['typography']['fontFamilies']['theme'];
+
+	$renamed_fonts = array(
+		'Alexandria'           => _x( 'Alexandria (Arabic)', 'Font name', 'jetpack' ),
+		'IBM Plex Sans Arabic' => _x( 'IBM Plex Sans (Arabic)', 'Font name', 'jetpack' ),
+		'Noto Sans Hebrew'     => _x( 'Noto Sans (Hebrew)', 'Font name', 'jetpack' ),
+		'Noto Sans HK'         => _x( 'Noto Sans (Hong Kong)', 'Font name', 'jetpack' ),
+		'Noto Sans JP'         => _x( 'Noto Sans (Japanese)', 'Font name', 'jetpack' ),
+		'Noto Sans KR'         => _x( 'Noto Sans (Korean)', 'Font name', 'jetpack' ),
+		'Noto Sans SC'         => _x( 'Noto Sans (Simplified Chinese)', 'Font name', 'jetpack' ),
+		'Noto Sans TC'         => _x( 'Noto Sans (Traditional Chinese)', 'Font name', 'jetpack' ),
+		'Noto Sans Telugu'     => _x( 'Noto Sans (Telugu)', 'Font name', 'jetpack' ),
+		'Noto Serif Hebrew'    => _x( 'Noto Serif (Hebrew)', 'Font name', 'jetpack' ),
+		'Noto Serif HK'        => _x( 'Noto Serif (Hong Kong)', 'Font name', 'jetpack' ),
+		'Noto Serif JP'        => _x( 'Noto Serif (Japanese)', 'Font name', 'jetpack' ),
+		'Noto Serif KR'        => _x( 'Noto Serif (Korean)', 'Font name', 'jetpack' ),
+		'Noto Serif SC'        => _x( 'Noto Serif (Simplified Chinese)', 'Font name', 'jetpack' ),
+		'Noto Serif TC'        => _x( 'Noto Serif (Traditional Chinese)', 'Font name', 'jetpack' ),
+	);
+
+	foreach ( $font_families as $key => $font_family ) {
+		$font_name = $font_family['name'];
+
+		if ( array_key_exists( $font_name, $renamed_fonts ) ) {
+			$font_families[ $key ]['name'] = $renamed_fonts[ $font_name ];
+		}
+	}
+
+	// See https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/#typography
+	$updated_fonts = array(
+		'version'  => 2,
+		'settings' => array(
+			'typography' => array(
+				'fontFamilies' => $font_families,
+			),
+		),
+	);
+
+	return $theme_json->update_with( $updated_fonts );
+}
+add_filter( 'wp_theme_json_data_theme', 'jetpack_rename_google_font_names', 10, 1 );
```

Comment:
Priority and arg count are default values and can be omitted.

Reply:
https://github.com/Automattic/jetpack/pull/27854/commits/55173b23d9e66ea3727cdf9e650281df9cf8f75a